cli/tests/test_git_clone.rs

```diff
@@ -429,48 +429,109 @@ fn test_git_clone_no_colocate() {
 
 #[test]
 fn test_git_clone_tags() {
-    let cases = [
-        (gix::remote::fetch::Tags::Included, [].as_slice()),
-        (
-            gix::remote::fetch::Tags::Included,
-            &["--fetch-tags", "included"],
-        ),
-        (gix::remote::fetch::Tags::All, &["--fetch-tags", "all"]),
-        (gix::remote::fetch::Tags::None, &["--fetch-tags", "none"]),
-    ];
-
-    insta::allow_duplicates! {
-        for (expected, args) in cases {
-            let test_env = TestEnvironment::default();
-            let root_dir = test_env.work_dir("");
-            let git_repo_path = test_env.env_root().join("source");
-            let _git_repo = git::init(git_repo_path);
-
-            // Clone an empty repo
-            let output = root_dir.run_jj(
-                [
-                    "git",
-                    "clone",
-                    "source",
-                    "repo",
-                    "--colocate"
-                ]
+    use gix::remote::fetch::Tags;
+
+    let test_env = TestEnvironment::default();
+    let root_dir = test_env.work_dir("");
+    let git_repo_path = test_env.env_root().join("source");
+    let source_git_repo = git::init(git_repo_path);
+
+    git::add_commit(
+        &source_git_repo,
+        "refs/tags/v1.0",
+        "foo",
+        b"content",
+        "message",
+        &[],
+    );
+
+    let commit = git::add_commit(
+        &source_git_repo,
+        "refs/tags/v2.0",
+        "bar",
+        b"content",
+        "message",
+        &[],
+    )
+    .commit_id;
+
+    git::add_commit(
+        &source_git_repo,
+        "refs/heads/main",
+        "baz",
+        b"content",
+        "message",
+        &[commit],
+    );
+
+    git::set_symbolic_reference(&source_git_repo, "HEAD", "refs/heads/main");
+
+    let run_test = |name, expected, args: &[_]| {
+        // Clone an empty repo
+        let output = root_dir.run_jj(
+            ["git", "clone", "source", name, "--colocate"]
                 .iter()
-                .chain(args)
-            );
-            insta::assert_snapshot!(output, @r#"
-            ------- stderr -------
-            Fetching into new repo in "$TEST_ENV/repo"
-            Nothing changed.
-            Hint: Running `git clean -xdf` will remove `.jj/`!
-            [EOF]
-            "#);
-
-            let git_repo = git::open(test_env.env_root().join("repo"));
-            let remote = git_repo.find_remote("origin").unwrap();
-            assert_eq!(expected, remote.fetch_tags());
-        }
-    }
+                .chain(args),
+        );
+
+        let git_repo = git::open(test_env.env_root().join(name));
+        let remote = git_repo.find_remote("origin").unwrap();
+        assert_eq!(expected, remote.fetch_tags());
+
+        output
+    };
```

Comment:
nit: I think this function can be split into two:
* `run(name, args) -> Output`
* `get_remote_fetch_tags() -> Tags`

Then, the expectation can be inlined in `assert_eq!()`, which will provide better diagnostics on error.